frog/private/posts.rkt

```diff
@@ -129,8 +129,15 @@
             #:when (member k '("Title" "Date" "Tags" "Authors"))
             (hash-set h (string-trim k) (string-trim v))]
            [s (warn s) h])))
+     (define d
+       (with-handlers ([exn:fail:contract?
```

Comment:
This is too specific and I think you should use `exn:fail?`. For example, when `date->date-struct` is given a bad format string, then its `match` will fail . `match` fails with `exn:fail?` so you won't catch that here.

(If as you mentioned you're concerned about hiding other bugs: Then I think the way to address that is to wrap only the call to `date->date-struct` in the `with-handlers`?)